xl2times/transforms.py

```diff
@@ -2152,6 +2166,65 @@ def make_str(df):
     return tables
 
 
+def _match_uc_wildcards(
+    df: pd.DataFrame,
+    process_map: dict[str, str],
+    dictionary: dict[str, pd.DataFrame],
+    matcher: Callable,
+    result_col: str,
+) -> pd.DataFrame:
+    """
+    Match wildcards in the given table using the given process map and dictionary.
+
+    Args:
+        df: Table to match wildcards in.
+        process_map: Mapping of column names to process sets.
+        dictionary: Dictionary of process sets to match against.
+        matcher: Matching function to use, e.g. get_matching_processes or get_matching_commodities.
+        result_col: Name of the column to store the matche results in.
+
+    Returns:
+        The table with the wildcard columns removed and the results of the wildcard matches added as a column named `results_col`
+    """
+    proc_cols = list(process_map.keys())
+
+    # drop duplicate sets of wildcard columns to save repeated (slow) regex matching.  This makes things much faster.
+    unique_filters = df[proc_cols].drop_duplicates().dropna(axis="rows", how="all")
```

Comment:
This is the main change here.  The previous logic was matching wildcard columns to process/commodity names for every row of wildcards.  Because the regex matching is slow, and the wildcards were often repeated many times, we now drop duplicates here, cutting down the number of regex matches by and order of magnitude or so.